File: VhallAndroidSDKLiveDemo/app/src/main/java/com/vhall/live/SplashActivity.java
package com.vhall.live;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;

import com.vhall.business.VhallSDK;
import com.vhall.uilibs.util.handler.WeakHandler;

/**
 * 程序启动页的Activity
 */
public class SplashActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_activity);
        new WeakHandler(new Handler.Callback() {
            @Override
            public boolean handleMessage(Message msg) {
                Intent intent;
                if (VhallSDK.isLogin()) {
                    intent = new Intent(SplashActivity.this, MainActivity.class);
                } else {
                    intent = new Intent(SplashActivity.this, LoginActivity.class);
                }
                startActivity(intent);
                finish();
                return false;
            }
        }).sendEmptyMessageDelayed(1, 2000);
    }
}

File: VhallAndroidSDKLiveDemo/app/src/main/java/com/vhall/live/SplashActivity.kt
package com.vhall.live

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.lifecycleScope
import com.vhall.business.VhallSDK
import com.vhall.live.databinding.ActivitySplashBinding
import com.vhall.uimodule.login.LoginActivity
import com.vhall.uimodule.main.MainActivity
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class SplashActivity : AppCompatActivity() {
    private val mViewBinding: ActivitySplashBinding by lazy {
        ActivitySplashBinding.inflate(
            layoutInflater
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(mViewBinding.root)
        lifecycleScope.launch {
            delay(2000)
            val intent: Intent = if (!VhallSDK.isLogin())
                Intent(this@SplashActivity, LoginActivity::class.java)
            else
                Intent(this@SplashActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }
}